pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/onchistory/OncHistoryUploadService.java

```diff
@@ -125,23 +129,38 @@ public void upload(String fileContent) {
     /**
      * Given participant short IDs in uploaded rows, verify the short ID, and get and record associated
      * participant IDs and medical record IDs
+     * verify if any exited participant(s) exists in passed onc history records.
      *
-     * @throws OncHistoryValidationException for failed verification
+     * @throws OncHistoryValidationException for failed verifications
      */
     protected Map<Integer, Integer> getParticipantIds(List<OncHistoryRecord> oncHistoryRecords,
                                                       ParticipantIdProvider participantIdProvider, boolean updateElastic) {
         Map<Integer, Integer> medIds = new HashMap<>();
+        List<String> exitedParticipants = new ArrayList<>();
 
         ParticipantDao participantDao = ParticipantDao.of();
 
+        String esIndex = new DDPInstanceDao().getDDPInstanceByInstanceName(realm).orElseThrow().getEsParticipantIndex();
         for (OncHistoryRecord rec : oncHistoryRecords) {
-            int participantId = participantIdProvider.getParticipantIdForShortId(rec.getParticipantTextId());
```

Comment:
I think there are tests that use `ParticipantIdProvider`. How are they affected with this removal?

Reply:
hmm . OncHistoryUploadServiceTest.java is setting and has ParticipantIdProvider code . 
Looks like all that need to be refactored now for the test to even run.  Not sure its all worth the time.

Reply:
Right.
If you look at my original suggestion it was to retain `ParticipantIdProvider` with a new `getParticipantIdForShortId` interface.

Are you suggesting disabling tests in  OncHistoryUploadServiceTest?

Reply:
refactored and added test case with EXITED participant